marlowe-playground-client/src/Language/Javascript/Interpreter.js

```diff
@@ -19,8 +19,8 @@ exports.eval_ = function (left, right, model) {
                           var slices = javascript.split(/^.*\/\* === Code above this comment will be removed at compile time === \*\/$/gm);
                           var takeSlice = 0;
                           if (slices.length > 1) { takeSlice = 1 };
-                          var justCode = slices.slice(takeSlice).join('');
-                          let res = safeEval(justCode, context);
+                          var justCode = 'function () { \n' + slices.slice(takeSlice).join('') + '\n if (typeof contract !== \'undefined\') { return contract } else { throw new Error("Variable \'contract\' not defined. Please, assign the resulting contract to the variable \'contract\'.") } }';
```

Comment:
are you 100% this will work, sometimes in JS you need to deal with lots of different types of null values. `undefined` `null` `something else` etc

Reply:
It seems it works if you forget to declare it. If you write `const contract = undefined` will do the same and if you write `const contract = null` it fails to parse the JSON. But I mean, if someone writes `const contract = null` they are not using the API we provide so they deserve it :)
Maybe we can add an explanation to the JSON error saying that you should use the API constructors provided and if that you did you should report the error

Reply:
the issue is that you get a confusing error if you declare `const contract = null;`.
```
There was an error when parsing the resulting JSON:
(NonEmptyList (NonEmpty (TypeMismatch "String" "Null") ((TypeMismatch "object" "object") : (TypeMismatch "object" "object") : (TypeMismatch "object" "object") : (TypeMismatch "object" "object") : (TypeMismatch "object" "object") : Nil)))
```

Reply:
Yes, so I will do that, also say:
`Please, use the JS API provided (see tutorial and examples). If you did use the JS API and still get this error, kindly report the problem at https://github.com/input-output-hk/plutus/issues/new and include the code that caused the error. Thank you`

Reply:
Oh! Now I see what you mean, this is the only case were we can get away without a type error, good point :man_facepalming: